### ISSUE TITLE
GitHub check for Google testing sometimes reverts to red after having been manually overridden to green

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#134684]
Internally filed as b/300299508.

on Sep 13, 2023, both @vashworth and myself experienced this on:

https://github.com/flutter/flutter/pull/134493
https://github.com/flutter/flutter/pull/134590
In both cases, what happened was:

Google testing fails, and the GitHub check UI reports that the tests reached the maximum number of retries
The PR author navigates to the FRoB dashboard, seeing the testing CL has been deleted
The PR author overrides the GitHub status
The PR GitHub check goes from red to green
Later, the check reverts back to red.

**Comment by christopherfujino** [comment]
When I try to update it again, I get:

```
generic::already_exists: Already exists: PremergeManualUpdateStatus(2fcb00915ee755e9f5c130daa32e0a3aa1b80fa7)
```

**Comment by XilaiZhang** [comment]
I think this was fixed. Casey volunteers to help follow up on this one and provide more contexts

**Comment by CaseyHillers** [comment]
There was a temporary fix in cl/565885771, which relied on a legacy database table for no longer updating a given PR. This had the edge case of new commits being pushed would skip Google testing, when the expectation is you would need to opt out again.

I sent a fix in cl/570368816. The issue was the wrong roll entry may have been updated, which wasn't the latest.
